### ISSUE TITLE
Support for socket options

### ISSUE THREAD

**Issue by anmonteiro** [ocaml-multicore/eio#322]
Right now both `Eio.Net.{connect,accept}` return a socket, but I didn't find a way to e.g. set `TCP_NODELAY` on the stream socket. Did I miss an API?

**Comment by talex5** [comment]
Looks like it's missing, indeed.

I was using it originally with the HTTP benchmark, but now that sends everything in a single packet it doesn't make any difference there.

As a work-around, you can of course get the FD directly:
```
Unix.setsockopt (Eio_unix.FD.peek_opt ssock |> Option.get) Unix.TCP_NODELAY true;
```

**Comment by create2000** [comment]
Hello @patricoferris . My name is Anthony Onah. I will be contributing on this issue. I hope with your guidance and support, i should be able to resolve this.

**Comment by talex5** [comment]
Note there is an old draft PR related to this at https://github.com/ocaml-multicore/eio/pull/575

**Comment by create2000** [comment]
@talex5. In rebasing it to the current main, are you proposing I start with the previous issue? Also, is it a good first issue? 

I will love to contribute.

**Comment by talex5** [comment]
Yes, I think that PR was going in the right direction. It looks like it just never got finished.

**Comment by rbjorklin** [comment]
Leaving a breadcrumb for any future folks who find their way here.

The way to do this now seems to be by using `Eio_unix.Net.fd` and `Eio_unix.Fd.use`.

**Comment by talex5** [comment]
Fixed in #859.
